app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,32 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build(micropost_params)
+    if @micropost.save
+      flash[:success] = t ".micropost_created"
+      redirect_to root_url
+    else
+      @feed_items = []
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+    @micropost.destroy
+    flash[:success] = t ".micropost_deleted"
+    redirect_to request.referrer || root_url
```

Comment:
You can use [https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back)